services/telemetry/telemetry.go

```diff
@@ -161,22 +161,38 @@ func (s *Service) prepareReport(ctx context.Context) (*reporter.ReportRequest, e
 			continue
 		}
 
+		// fetch metrics from DS
+		metricFetchStartTime := time.Now()
 		metrics, err := ds.FetchMetrics(ctx, telemetry)
+		metricFetchTook := time.Since(metricFetchStartTime)
+		s.l.Debugf("fetching [%s] took [%s]", telemetry.ID, metricFetchTook)
+		totalTime += metricFetchTook
 		if err != nil {
 			s.l.Debugf("failed to extract metric from datasource for [%s]:[%s]: %v", telemetry.Source, telemetry.ID, err)
 			continue
 		}
 
-		current.Metrics = append(current.Metrics, metrics...)
+		for _, each := range metrics {
+			telemetryMetric, err := s.makeMetric(ctx)
+			if err != nil {
+				s.l.Debugf("failed to make Metric %v", err)
+				continue telemetryLoop
```

Comment:
it can be just `break`

Reply:
in need to jump to outer loop, break will not work here

Reply:
there are no operations after inner loop, so break will do the same.